Cyborgs now use storage datum

JohnFulpWillard opened the pull request:
## About The Pull Request

This moves Cyborgs onto using storage datums, removing the remenants of the shitcode that was Cyborg inventory. It's now done mostly by equipping/unequipping/storage items, much like how other mobs do.
This allows borgs to take advantage of more hand support stuff and things like ``dropped()``, so borgs no longer have to copy paste drop code to ``cyborg_unequip``

It also:
- Removes ``CYBORG_ITEM_TRAIT``
- Removes all borg items being ``NODROP``

https://github.com/user-attachments/assets/11442a10-3443-41f2-8c72-b38fb0126cdb

## Why It's Good For The Game

Currently borgs are able to have their entire inventory open and a bag below it, which I thought was a little weird. I always assumed they WERE storage items, so I guess I'm doing it myself.
Cyborgs using storage code makes it easier for contributors to actually do stuff with, without risking breaking everything. It also hopefully will make borg items more resilient against breaking in the future, now that we're not relying on nodrop.
Also just brings them more in line with other mobs, all of which make use of storages.

## Changelog

:cl:
refactor: Cyborg's modules now use storage (so opening a bag will close modules instead of overlap one over the other).
qol: Observers can now see Cyborg's inventories (like they can for humans).
/:cl:

JohnFulpWillard committed:
Fixes HUDs for people with 4 hands

JohnFulpWillard committed:
remove carbon check

JohnFulpWillard committed:
Exclude cyborgs

JohnFulpWillard committed:
pt 1, still broken asf

JohnFulpWillard committed:
Update storage_interface.dm

JohnFulpWillard committed:
Storage is now static

JohnFulpWillard committed:
more minor code improvements

JohnFulpWillard committed:
update inventory when emagged/upgraded

JohnFulpWillard committed:
Removes advanced cyborg mops too

JohnFulpWillard committed:
Can use only silicon storage

MrMelbert commented:
We can probably kill `cyborg_unequip` entirely and just make it a comsig sent from the storage datum, which is pretty neat

optimumtact commented:
this will need a test merge

JohnFulpWillard committed:
Merge remote-tracking branch 'upstream/master' into cyborg-inventory

JohnFulpWillard committed:
items stay in hand and and say you can't drop it

JohnFulpWillard commented on code/datums/storage/storage.dm:
```diff
@@ -369,7 +371,7 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
  */
 /datum/storage/proc/can_insert(obj/item/to_insert, mob/user, messages = TRUE, force = STORAGE_NOT_LOCKED)
-	if(QDELETED(to_insert) || !istype(to_insert))
+	if(QDELETED(to_insert) || !istype(to_insert) || iscyborg(user))
```
> could get rid of the weird cyborg blocker in item_interact_insert then too.

What are you talking about here?

SmArtKar replied:
> > could get rid of the weird cyborg blocker in item_interact_insert then too.
> 
> What are you talking about here? I would assume the line you're commenting on but the message makes it sound like it's talking about a separate instance of it

I'm saying that if you implement my suggestion, then another instance of a cyborg check could be removed

JohnFulpWillard committed:
a

JohnFulpWillard committed:
bandaid support for observers

A lot of support for observers watching hand items (update hand icons and such) is tied to the carbon annoyingly. This puts a bandaid of bringing back the old solution to ensure observers can see it, and hopefully in the future someone can bring the carbon procs down to living or something.

Maurukas reviewed (approved):
Accidentally mapped cyborg part, neat.

SmArtKar commented:
LGTM but needs TMing

JohnFulpWillard committed:
Merge branch 'master' into cyborg-inventory

JohnFulpWillard committed:
Update storage_interface.dm

JohnFulpWillard committed:
small update

JohnFulpWillard committed:
Update inventory.dm

LemonInTheDark commented:
did this get testmerged?

JohnFulpWillard committed:
Testmerge fixes

JohnFulpWillard committed:
Merge branch 'master' into cyborg-inventory

JohnFulpWillard committed:
model check too

JohnFulpWillard committed:
Merge branch 'master' into cyborg-inventory

JohnFulpWillard committed:
revert opacity changes

JohnFulpWillard committed:
Borgs cant put their omnitools in autolathes

Ghommie committed:
Merge branch 'master' into cyborg-inventory
